### PR TITLE
Check memory

### PR DESCRIPTION
**Pull request by ahoyosid**
With this PR we solve the issue #1365.
We separate the function that checks the memory in _utils/chache_mixin.py, and use it use it in check_embedded_nifti_masker.

Basically this functions returns a memory object with the predefine cachedir.

Ping @GaelVaroquaux  @KamalakerDadi

### CODE REVIEW

**Comment by bthirion** [comment]
If you could explain a bit more what is tested, that would help. The PR looks good otherwise.

File: nilearn/_utils/cache_mixin.py [hunk 1]
```diff
@@ -30,6 +30,46 @@
 __CACHE_CHECKED = dict()
 
 
+def _check_memory(memory, verbose=False):
```
Comment (GaelVaroquaux) [inline nilearn/_utils/cache_mixin.py]: A tiny docstring would really help. In particular detailing the accepted types of the memory argument, as this is non trivial.

**Comment by GaelVaroquaux** [comment]
This looks good to me. Thanks. Will merge once you do the docstring.

File: nilearn/_utils/cache_mixin.py [hunk 2]
```diff
@@ -30,7 +30,23 @@
 __CACHE_CHECKED = dict()
 
 
-def _check_memory(memory, verbose=False):
+def _check_memory(memory, verbose=0):
+    """Function to make ensure a memory object.
+
+    Parameters
+    ----------
+    memory: instance of joblib.Memory or string
```
Comment (GaelVaroquaux) [inline nilearn/_utils/cache_mixin.py]: The function also accepts None, doesn't it?

**Comment by GaelVaroquaux** [comment]
Good to go. Merging. Thanks

File: nilearn/_utils/cache_mixin.py [hunk 3]
```diff
@@ -30,6 +30,62 @@
...
+    """Function to make ensure a memory object.
+
+    Parameters
+    ----------
+    memory: None or instance of joblib.Memory or string
+        Used to cache the masking process.
+        If a string is given, it is the
+        path to the caching directory.
+
+    verbose : int, optional (default 1)
```
Comment (KamalakerDadi) [inline nilearn/_utils/cache_mixin.py]: default is 0 but you mentioned as 1

Comment (ahoyosid) [inline nilearn/_utils/cache_mixin.py]: Oh, that's true...
It's is already merged. Do you know how can I change this?

File: nilearn/_utils/cache_mixin.py [hunk 4]
```diff
@@ -30,6 +30,62 @@
 __CACHE_CHECKED = dict()
 
 
+def _check_memory(memory, verbose=0):
+    """Function to make ensure a memory object.
```
Comment (KamalakerDadi) [inline nilearn/_utils/cache_mixin.py]: Function to make ensure an instance of joblib memory object.

File: nilearn/_utils/cache_mixin.py [hunk 5]
```diff
@@ -30,6 +30,62 @@
...
+
+    Parameters
+    ----------
+    memory: None or instance of joblib.Memory or string
```
Comment (KamalakerDadi) [inline nilearn/_utils/cache_mixin.py]: May prefer to denote "str" rather than string since you denote integer as int below. For the sake of consistency.

**Comment by GaelVaroquaux** [comment]
> It's is already merged. Do you know how can I change this?

Send a new PR.

**Pull request merged**